File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/BrandService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbBrand;

import java.util.List;

public interface BrandService {
    /**
     * 条件查询
     * @param pageNum
     * @param pageSize
     * @param brand
     * @return
     */
    public PageResult getBrandPage(int pageNum,int pageSize,TbBrand brand);
    /**
     * 添加品牌
     * @param brand
     */
    public void addBrand(TbBrand brand);

    /**
     * 根据id查询品牌信息
     * @param id
     * @return
     */
    public TbBrand getBrandById(long id);

    /**
     * 修改品牌信息
     * @param brand
     */
    public void updateBrand(TbBrand brand);

    /**
     * 删除集合中对应id值的品牌
     * @param ids
     */
    public void deleteBrand(Long[] ids);
}

File: youlexuan-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {
    /**
     * 条件查询
     * @param pageNum
     * @param pageSize
     * @param searchEntity
     * @returns {*}
     */
    this.getBrandPage=function (pageNum,pageSize,searchEntity) {
        return $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity)
    }
    /**
     * 添加品牌
     * @param brand
     */
    this.addBrand=function (brand) {
        return $http.post("../brand/addBrand.do",brand)
    }
    /**
     * 修改品牌
     * @param brand
     */
    this.updateBrand=function (brand) {
        return $http.post("../brand/updateBrand.do",brand)
    }
    /**
     * 根据id查询品牌
     * @param id
     */
    this.getBrandById=function (id) {
        return $http.post("../brand/getBrandById.do?id="+id)
    }
    /**
     * 删除品牌
     * @param selectionIds
     */
    this.deleteBrand=function (selectionIds) {
        return $http.get("../brand/deleteBrand.do?ids="+selectionIds)
    }
})